jellyfish/_notifier.py

```diff
@@ -0,0 +1,130 @@
+'''
+Notifier listening to WebSocket events
+'''
+
+import asyncio
+from typing import Callable, Any
+
+import betterproto
+
+from websockets import client
+from websockets.exceptions import ConnectionClosed
+
+from jellyfish.events import (
+    ServerMessage, ServerMessageAuthRequest, ServerMessageAuthenticated,
+    ServerMessageSubscribeRequest, ServerMessageEventType, ServerMessageSubscribeResponse,
+    ServerMessageMetricsReport)
+
+
+class Notifier:
+    '''
+    Allows for receiving WebSocket messages from Jellyfish.
+    '''
+
+    def __init__(self,
+                 server_address: str = 'localhost:5002', server_api_token: str = 'development'):
+        self._server_address = server_address
+        self._server_api_token = server_api_token
+        self._websocket = None
+        self._ready = False
+
+        self._ready_event: asyncio.Event = None
+
+        self._notification_handler: Callable = None
+        self._metrics_handler: Callable = None
+
+    def on_server_notification(self, handler: Callable[[Any], None]):
+        '''
+        Decorator used for defining handler for ServerNotifications
+        i.e. all messages other than `ServerMessageMetricsReport`.
+        '''
+        self._notification_handler = handler
+        return handler
+
+    def on_metrics(self, handler: Callable[[ServerMessageMetricsReport], None]):
+        '''
+        Decorator used for defining handler for `ServerMessageMetricsReport`.
+        '''
+        self._metrics_handler = handler
+        return handler
+
+    async def connect(self):
+        '''
+        A coroutine which connects Notifier to Jellyfish and listens for all incoming
+        messages from the Jellyfish.
+
+        It runs until the connection isn't closed.
+
+        The incoming messages are handled by the functions defined using the
+        `on_server_notification` and `on_metrics` decorators.
+        '''
+        async with client.connect(f'ws://{self._server_address}/socket/server/websocket') \
+                as websocket:
```

Comment:
Nitpick, but I don't like the line break
```suggestion
        address = f'ws://{self._server_address}/socket/server/websocket'
        async with client.connect(address) as websocket:
```